Zhuoxiao Chen - 44815404 - StyleGAN - OAI AKOA Knee dataset

zhuoxiao-chen opened the pull request:
Hello COMP3710 team,

Please review my pull request containing my implementation of StyleGAN as applied to the OAI AKOA Knee dataset. My repo contains several python scripts and a markdown file, as simply explained below: 

**network.py**: contains discriminator and generator and all the modules and components required to build the discriminator and generator.

**data_loader.py**: a PyTorch dataset to be loaded that supports all dimensions for each AKOA image. 

**preprocess_AKOA_images.py**: the AKOA pre-processing method to convert each AKOA image to all the different required dimensions (8 times 8, 16 times 16, .., 256 times 256)

**train_AKOA_StyleGAN.py**: the training process, which accepts pre-processed AKOA images to train the StyleGAN. 

**generate_AKOA.py**: to produce fake AKOA images using trained StyleGAN.

**README.md**: A report on the work containing project description, algorithm explanation, usage of python scripts, implementation details and result visualisation. 

Regards,
Zhuoxiao Chen
44815404

Commit:
Initial commit by Zhuoxiao.

Commit:
Add initial Pycharm files by Zhuoxiao

Commit:
Add .gitignore to stop uploading datasets to the GitHub Repo

Commit:
Fix .gitignore to stop uploading datasets to the GitHub Repo

Commit:
Rename the train.py to main.py; add another .gitignore to the folder of my work

zhuoxiao-chen committed:
import required libraries, set up required arguments by argparse.

zhuoxiao-chen committed:
update .gitignore

zhuoxiao-chen committed:
complete data_loader to load datasets with various resolutions.

zhuoxiao-chen committed:
complete Discriminator in network.py

zhuoxiao-chen committed:
complete Generator Part in network.py; complete model and optimiser set up in train.py

zhuoxiao-chen committed:
Add some comments for function PixelNorm in network.py

zhuoxiao-chen committed:
Add some comments in Network.py for code readers understand the code easily.

zhuoxiao-chen committed:
Fix some bugs in Discriminator and make some comments.

zhuoxiao-chen committed:
Fix some bugs in Discriminator and make some comments.

zhuoxiao-chen committed:
Update README.md

Set up a basic structure of the readme documentation.

zhuoxiao-chen committed:
add image transformormation to the train file

zhuoxiao-chen committed:
Training section completed, now we can start to train the network.

zhuoxiao-chen committed:
Writing the documentation - README.md

zhuoxiao-chen committed:
Complete the Introduction of README.md

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
Add some links to the introduction

zhuoxiao-chen committed:
Add some links to this report.

zhuoxiao-chen committed:
Add link to CVPR2019

zhuoxiao-chen committed:
Insert model figure of styleGAN

zhuoxiao-chen committed:
Adjust the figure size

zhuoxiao-chen committed:
Adjust Figure size

zhuoxiao-chen committed:
Adjust Figure size

zhuoxiao-chen committed:
Writing the documentation part.

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
Writing the MD

zhuoxiao-chen committed:
Writing MD

zhuoxiao-chen committed:
Writing the MD

zhuoxiao-chen committed:
Write the MD

zhuoxiao-chen committed:
Writing MD

zhuoxiao-chen committed:
Delete recognition/s4481540_Zhuoxiao_Chen/.idea directory

zhuoxiao-chen committed:
Write MD, Standard GAN part

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
Writing Progressive GAN

zhuoxiao-chen committed:
Add figure of Progressive GAN

zhuoxiao-chen committed:
Add more functions to the train file, as well as the comments

zhuoxiao-chen committed:
Complete the generation part, now we can generate the images using the saved trained checkpoint

zhuoxiao-chen committed:
Add some generated images

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
complete a function - change_gradient_status() to freeze a network.

zhuoxiao-chen committed:
Merge branch 'topic-recognition' of https://github.com/zhuoxiao-chen/PatternFlow into topic-recognition

zhuoxiao-chen committed:
complete a function - change_gradient_status() to freeze a network.

zhuoxiao-chen committed:
complete a function learning_rate_decay().

zhuoxiao-chen committed:
Make the code in train() more clear

zhuoxiao-chen committed:
train function is improved

zhuoxiao-chen committed:
writing dataset section

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
writing the dataset section

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
add one figure of original image

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
Add some requirements

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
add requirements as code block to the MD

zhuoxiao-chen committed:
Make some important information bold

zhuoxiao-chen committed:
Writing Implementation Details

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
add some meaningful comments to the train file

zhuoxiao-chen committed:
add the data pre-processing file to the project

zhuoxiao-chen committed:
Merge branch 'topic-recognition' of https://github.com/zhuoxiao-chen/PatternFlow into topic-recognition

zhuoxiao-chen committed:
update the data pre-processing file to the project

zhuoxiao-chen committed:
some good comments added to the train file

zhuoxiao-chen committed:
some good comments added to the train file

zhuoxiao-chen committed:
some good comments added to the train file

zhuoxiao-chen committed:
make some explanation to the saving model section of train file

zhuoxiao-chen committed:
complete model explanation in the most important train() function

zhuoxiao-chen committed:
organise the parameters for better usage

zhuoxiao-chen committed:
A bug was found, fixed from previous checkpoint

zhuoxiao-chen committed:
update the parameter information for training.

zhuoxiao-chen committed:
Finally complete the train.py file with full comments, and unused libraries are removed.

zhuoxiao-chen committed:
Writing the usage of train function

zhuoxiao-chen committed:
update data pre-process file in main function and improve the accepted parameters.

zhuoxiao-chen committed:
update data pre-process file for image_resize_multiple function.

zhuoxiao-chen committed:
update data pre-process script.

zhuoxiao-chen committed:
writing the instruction for pre-processing

zhuoxiao-chen committed:
add more parameter information

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
A bug fixed when generating the AKOA image

zhuoxiao-chen committed:
Merge branch 'topic-recognition' of https://github.com/zhuoxiao-chen/PatternFlow into topic-recognition

zhuoxiao-chen committed:
A bug fixed when generating the AKOA image; remove generater images

zhuoxiao-chen committed:
Update the parameters information in generate_AKOA.py

zhuoxiao-chen committed:
writing how to use generate_AKOA.py

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
writing Implementation

zhuoxiao-chen committed:
Add some images of 20k iterations

zhuoxiao-chen committed:
Complete the conclusion

zhuoxiao-chen committed:
Update README.md

zhuoxiao-chen committed:
fix some typo in readme

zhuoxiao-chen committed:
fix some type

zhuoxiao-chen committed:
Update some meaningful comments in network.py

zhuoxiao-chen committed:
Update data_loader documentation

WeiDaiwow reviewed (commented):
1. Have both test and driver file
2. Good commits 
3. Nice readme file
4. Algorithm solves the problems
5. Good design and nice implementation
6. Nice comments in the code
7. Nice description and explanation of the algorithm
8. Correct pull request with the description
9. nice visualization 

Suggestion:
1. Please provide more details about how images are generated during the testing stage in the readme file. (What is the input for each model and what is the output)
2. Please provide some plots during the training such as a convergence plot for both generators and discriminators
3. Please fix the markdown "Email: zhuoxiao [dot] chen [dot] uq [dot] edu [dot] au"
4. Please provide reference as a list within the readme file

zhuoxiao-chen committed:
Fixed the typo in the email.

zhuoxiao-chen committed:
More description of generate_AKOA.py in READ.md

More details about how the generate_AKOA.py script works are given, and the input of the generator and the output saved images are specified.

zhuoxiao-chen committed:
Add reference list to the READ.me

zhuoxiao-chen committed:
Improve reference list

shakes76 commented:
## TF/Torch Usage
Adequate use and implementation
## Good Practice (Design/Commenting)
Good spacing and comments
No Header blocks -1
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present 
No Data leakage 
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch)
No feedback required
Request Description good
